android/app/src/main/java/network/mysterium/ui/AccountViewModel.kt

```diff
@@ -0,0 +1,155 @@
+        val channelAddress: String,
+        var status: IdentityRegistrationStatus
+) {
+    val registered: Boolean
+        get() {
+            return status == IdentityRegistrationStatus.REGISTERED_CONSUMER
+        }
+}
+
+class BalanceModel(val balance: TokenModel)
+
+class TokenModel(token: Long = 0) {
+    var displayValue = ""
+    var value = 0.00
+
+    init {
+        value = token / 100_000_000.00
+        displayValue = "%.3f MYSTT".format(value)
+    }
+}
+
+class AccountViewModel(private val nodeRepository: NodeRepository, private val bugReporter: BugReporter) : ViewModel() {
+    val balance = MutableLiveData<BalanceModel>()
+    val identity = MutableLiveData<IdentityModel>()
+
+    suspend fun load() {
+        initListeners()
+        loadIdentity()
+    }
+
+    suspend fun topUp() {
+        try {
+            val currentIdentity = identity.value ?: return
+            nodeRepository.topUpBalance(currentIdentity.address)
+        } catch (e: Exception) {
+            Log.e(TAG, "Failed to top-up balance", e)
+        }
+    }
+
+    fun needToTopUp(): Boolean {
+        if (balance.value == null) {
+            return false
+        }
+        return balance.value!!.balance.value < 0.01
+    }
+
+    fun isIdentityRegistered(): Boolean {
+        val currentIdentity = identity.value ?: return false
+        return currentIdentity.registered
+    }
+
+    private suspend fun initListeners() {
+        nodeRepository.registerBalanceChangeCallback {
+            handleBalanceChange(it)
+        }
+
+        nodeRepository.registerIdentityRegistrationChangeCallback {
+            handleIdentityRegistrationChange(it)
+        }
+    }
+
+    private fun handleIdentityRegistrationChange(it: String) {
+        val currentIdentity = identity.value ?: return
+        viewModelScope.launch {
+            currentIdentity.status = IdentityRegistrationStatus.parse(it)
+            identity.value = currentIdentity
+        }
+    }
+
+    private fun handleBalanceChange(it: Balance) {
+        viewModelScope.launch {
+            balance.value = BalanceModel(TokenModel(it.value))
+        }
+    }
+
+    private suspend fun loadIdentity() {
+        try {
+            // Load node identity and it's registration status.
+            val nodeIdentity = nodeRepository.getIdentity()
+            val identityResult = IdentityModel(
+                    address = nodeIdentity.address,
+                    channelAddress = nodeIdentity.channelAddress,
+                    status = IdentityRegistrationStatus.parse(nodeIdentity.registrationStatus)
+            )
+            identity.value = identityResult
+            bugReporter.setUserIdentifier(nodeIdentity.address)
+            Log.i(TAG, "Loaded identity ${nodeIdentity.address}, channel addr: ${nodeIdentity.channelAddress}")
+
+            if (identityResult.status == IdentityRegistrationStatus.UNREGISTERED || identityResult.status == IdentityRegistrationStatus.REGISTRATION_ERROR) {
```

Comment:
I suggest importing because that will make the code more concise.
```kt
import network.mysterium.ui.IdentityRegistrationStatus.*

...

if (setOf(UNREGISTERED, REGISTRATION_ERROR).contains(identityResult.status)) {
     registerIdentity()
}
```